addons/website_slides/models/slide_channel.py

```diff
@@ -285,18 +363,32 @@ def _compute_has_requested_access(self):
         for channel in self:
             channel.has_requested_access = channel.id in requested_cids
 
-    @api.depends('channel_partner_ids.partner_id')
+    @api.depends('channel_partner_ids.partner_id', 'channel_partner_ids.is_pending')
     @api.depends_context('uid')
     @api.model
     def _compute_is_member(self):
         channel_partners = self.env['slide.channel.partner'].sudo().search([
             ('channel_id', 'in', self.ids),
+            ('is_pending', '=', False),
```

Comment:
Looks like "is_member" is not aligned anymore with "member_count", which is a bit strange.

Maybe we should rename for more clarity to:

"is_member"
"is_member_pending"
"members_count"
"members_pending_count"

Being a member means you have accepted and thus not pending anymore.
And the members_count would count all the members based on the same condition.

As for your "allowed_member" thing, why not simply turning it into "is_member" OR "is_member_pending" when we need it? Instead of introducing an extra field.

Purpose is to have a clear structure.

What do you think?

Side note: I think we could compute members_count and members_pending_count in the same compute method by grouping on "channel_id" AND "is_pending".
(You will probably have to turn the lazy mode off if you group by multiple fields, I think it's simply passing lazy=False in the method params).

Reply:
I can indeed remove the variable in python. However, when allocating data for widgets in the xml I still use isAllowedMember = is_member or is_member_pending, to make it shorter in js files - I updated the definition. Elsewhere, I use "is_member or is_member_pending" instead in the python code. I can still change the js name variable though?

About side note : That works, but could we not do the same with member_done_count and member_active_count ? grouping on both channel_id and completed, and adding is_invitation_pending = False in the domain ? 
We would have two compute methods for the 4 counts. _compute_members_counts_1 and _compute_members_counts_2 (or more explicit method names)

Reply:
You could create a single _compute_members_counts method and do both requests within it, if that's an option.

I let you play around with it and see what's best.
Goal is to have a good combo between performance / readability.